scripts/generateForDay.sh

```diff
@@ -214,7 +214,7 @@ if [ "${DO_TIMELAPSE}" = "true" ] ; then
 fi
 
 
-if [ "${TYPE}" = "GENERATE" -a ${SILENT} = "false" ]; then
+if [ "${TYPE}" = "GENERATE" -a ${SILENT} = "false" -a ${EXIT_CODE} -eq 0 ]; then
```

Comment:
EXIT_CODE generation is a little bit confusing and result is maybe wrong - but not critical
```
pi@allsky:~/allsky $ A=0
pi@allsky:~/allsky $ B=1
pi@allsky:~/allsky $ C=-1
pi@allsky:~/allsky $ echo $((${A}+${B}+${C}))
0
```

-a is depricated (https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html)
```
The XSI extensions specifying the -a and -o binary primaries and the '(' and ')' operators have been marked obsolescent. (Many expressions using them are ambiguously defined by the grammar depending on the specific expressions being evaluated.) Scripts using these expressions should be converted to the forms given below. Even though many implementations will continue to support these obsolescent forms, scripts should be extremely careful when dealing with user-supplied input that could be confused with these and other primaries and operators. Unless the application developer knows all the cases that produce input to the script, invocations like:

test "$1" -a "$2"

should be written as:

test "$1" && test "$2"

to avoid problems if a user supplied values such as $1 set to '!' and $2 set to the null string. That is, in cases where maximal portability is of concern, replace:

test expr1 -a expr2
```